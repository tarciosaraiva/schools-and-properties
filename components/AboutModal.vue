<template>
  <modal @close="closeFn">
    <template #header>
      <h3>About</h3>
    </template>
    <template #body>
      <p>The information provided by this website is a collation of 3 major data sources, listed in the attributions modal accessed by the link at the footer of this website.</p>
      <p>We are showing all schools currently listed in the Data Vic. The ones faded do not have any rating associated with them. The ones not faded are the ones that have a score of 90 or above in <a href="https://www.bettereducation.com.au/" target="blank">Better Education</a> website so.</p>
      <p>You will also notice that the not faded schools have different sizes: this is due to your rating filter. If you have a score of e.g. 95 then every school that matches that number or above will have a larger icon than the ones that have a lower ranking.</p>
      <p>
        <a href="https://www.buymeacoffee.com/tarciosaraiva" target="blank"><small>Support my work</small></a>
      </p>
      <p>
        <a href="https://www.github.com/tarciosaraiva/schools-and-properties" target="blank"><small>Project on Github</small></a>
      </p>
      <p>
        <a href="https://forms.gle/7xZTuGrb3w7VAUdq6" target="blank"><small>Give us feedback</small></a>
      </p>
    </template>
  </modal>
</template>

<script lang="ts">
import Vue from 'vue'
import Modal from '~/components/Modal.vue'

export default Vue.extend({
  components: { Modal },
  props: {
    closeFn: Function
  }
})
</script>
