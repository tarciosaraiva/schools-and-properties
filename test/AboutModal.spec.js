import { mount } from '@vue/test-utils'
import AboutModal from '~/components/AboutModal.vue'

describe('AboutModal', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(AboutModal, { propsData: { closeFn: jest.fn() } })
  })

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
