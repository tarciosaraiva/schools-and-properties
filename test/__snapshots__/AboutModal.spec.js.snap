// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutModal renders correctly 1`] = `
<transition-stub name="modal">
  <div class="modal-mask">
    <div class="modal-wrapper">
      <div class="modal-container">
        <div class="modal-header">
          <h3>About</h3>
        </div>
        <div class="modal-body">
          <p>The information provided by this website is a collation of 3 major data sources, listed in the attributions modal accessed by the link at the footer of this website.</p>
          <p>We are showing all schools currently listed in the Data Vic. The ones faded do not have any rating associated with them. The ones not faded are the ones that have a score of 90 or above in <a href="https://www.bettereducation.com.au/" target="blank">Better Education</a> website so.</p>
          <p>You will also notice that the not faded schools have different sizes: this is due to your rating filter. If you have a score of e.g. 95 then every school that matches that number or above will have a larger icon than the ones that have a lower ranking.</p>
          <p>
            <a href="https://www.buymeacoffee.com/tarciosaraiva" target="blank">
              <small>Support my work</small>
            </a>
          </p>
          <p>
            <a href="https://www.github.com/tarciosaraiva/schools-and-properties" target="blank">
              <small>Project on Github</small>
            </a>
          </p>
          <p>
            <a href="https://forms.gle/7xZTuGrb3w7VAUdq6" target="blank">
              <small>Give us feedback</small>
            </a>
          </p>
        </div>
        <div class="modal-footer text-right">
          <button class="pl-button full small">
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</transition-stub>
`;
